app/controllers/api/restful_controller.rb

```diff
@@ -116,31 +116,35 @@ def resource_symbol
   end
 
   def resource_name
-    @@resource_name || controller_name.singularize
+    controller_name.singularize
   end
 
   def resource_class
-    @@resource_class || resource_name.camelize.constantize
+    resource_name.camelize.constantize
   end
 
   def resource_plural
-    @@resource_plural || controller_name
+    controller_name
+  end
+
+  def resource_serializer
+    "#{resource_name}_serializer".camelize.constantize
   end
 
   def service
-    @@service_class || "#{resource_name}_service".camelize.constantize
+    "#{resource_name}_service".camelize.constantize
   end
 
-  def respond_with_collection(scope: {})
-    render json: collection, root: serializer_root, scope: scope
+  def respond_with_collection(scope: {}, serializer: resource_serializer)
+    render json: collection, root: serializer_root, scope: scope, each_serializer: serializer
   end
 
-  def respond_with_resource
+  def respond_with_resource(scope: {}, serializer: resource_serializer)
```

Comment:
I merged in the stuff from the popolo motion endpoint to get this ability to inject what serializer to use, which comes in handy for using CurrentUserSerializer over UserSerializer for the profile page.